### ISSUE TITLE
Merging of overlapping unitigs

### ISSUE THREAD

**Issue by olekto** [nadegeguiglielmoni/GraphUnzip#8]
Hi,
I noticed that when testing GraphUnzip, the two resulting haplotypes are larger than what hifiasm produce on its own. I also noticed that in your preprint you write that for potato "The draft assembly was 1.51 Gb, and after phasing with GraphUnzip, the assembly size rose to 1.67-1.73 Gb. " 

Why would the assembly size increase? You don't insert any new sequences as far as I understand (the ONT reads are not used for gap-closing for instance). Is the increase due to not trimming one of the unitigs when two overlapping unitigs are merged? 

I tried looking at the code to see how you handle overlapping unitigs, but I couldn't see that exactly. However, I see that in this code you state that you don't modify the input unitigs, but I couldn't see what you exactly do: https://github.com/nadegeguiglielmoni/GraphUnzip/blob/e3d05a163ce508050139b6e838c5a0d16c3cf520/segment.py#L438

If two unitigs overlap by, for instance, 10000 bp (10000M in the CIGAR), then one of the unitigs should be trimmed. Or am I mistaken? What do you do in these cases?

Thank you.

Sincerely,
Ole

**Comment by RolandFaure** [comment]
Hello Ole,
The increase in size is not related to the overlaps.
The increase in size comes from the duplication of the contigs that are actually present in multiple copies in the genome but only once in the assembly. Typically, in diploid organisms, homozygous regions are present twice in the genome (once on each chromosome). Thus they should be present twice on a perfect end-to-end assembly. Yet, because an homozygous region is by definition identical on the two chromosomes, it is often only assembled in one copy in the assemblies. GraphUnzip then needs to duplicate that region to try to recover the original chromosomes. 
To sum up, the original assembly is generally smaller than the expected size for a full phased assembly: the assembly of a 2 * 3Gb human will yield, for example, a 4Gb assembly. The size will then increase during GraphUnzip to reach, hopefully, a finished 6Gb diploid assembly.

As for overlaps, they are indeed trimmed when producing final merged contigs, but that is done during the output of the program, as GraphUnzip does not handle the sequences themselves except during import/export (a somewhat messy procedure in the output_to_gfa(merge=True) function).

Yours,
Roland
